lib/db_connection/connection.ex

```diff
@@ -169,23 +169,25 @@ defmodule DBConnection.Connection do
         mon = Process.monitor(pid)
         handle_checkout({ref, mon}, timeout, from, state, s)
       %{client: :closed} ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of disconnection"
+        err = DBConnection.Error.exception(message)
         {:reply, {:error, err}, s}
       %{queue: queue} when queue? == true ->
         client = {ref, Process.monitor(pid)}
         queue = :queue.in({client, timeout, from}, queue)
         {:noreply, %{s | queue: queue}}
       _ when queue? == false ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of queue"
```

Comment:
It is awkward to say "connection already checked out" because we want to return this error from the pools too, where there can be many connections and none are available.